composer/algorithms/selective_backprop/selective_backprop.py

```diff
@@ -272,6 +272,8 @@ def apply(self, event: Event, state: State, logger: Optional[Logger] = None) ->
                     raise RuntimeError('Model must be of type ComposerModel')
                 self._loss_fn = state.model.loss
             return
+
+        state.batch = state.device.batch_to_device(state.batch)
```

Comment:
why is this necessary?

Reply:
selective backprop forwards the whole batch to determine what fraction of the minibatch to use for gradient calc - unit tests also break otherwise